### DISCUSSION TITLE
Combined Drawer and Dialog - Standalone Component Proposal

### DISCUSSION THREAD

**Discussion by lukasjoho** [shadcn-ui/ui#2669] [Ideas]
Hey ShadCN-lers 👋 (builders and consumers), 

I would like to offer merging the `<Drawer/>` and `<Dialog/>` component into an additional standalone **"Responsive Modal"** component (name tbd).

![modal](https://github.com/shadcn-ui/ui/assets/53558967/27faf91f-3f90-43e1-be51-f13ae929efba)

✨ This will make it so much easier building dialog content for mobile! **One component instead of stitching two together!** ✨

**Situation** 
Currently a "Responsive Dialog" is shown as an [example on the ShadCN website](https://ui.shadcn.com/docs/components/drawer#:~:text=Examples-,Responsive%20Dialog,-You%20can%20combine) to combine a `<Drawer/>` (on mobile) with a `<Dialog/>` (on desktop). 

And this is how modals should work. A `<Drawer/>`  will most often be preferred over a `<Dialog/>` on mobile. 
However, at this point, one then needs to copy the example and pass content into both components. 

> Why not build a wrapper around both to offer an easier interface?

As both `<Drawer/`> and `<Dialog/>` follow the builder pattern of `<... .Provider/>`, `<... .Trigger/>`, `<... .Content/>`, `<... .Footer/>` etc... it would be very much feasible to bring components of both `<Drawer/>` and `<Dialog/>` into one unified interface. 

Use that new `<ResponsiveDialog/>` (imagine a nicer name) component and you'll automatically get different Triggers, Footers, Content, etc... for mobile and for desktop. 

Would be keen to understand whether this is not just me, but whether it is actually of higher demand! 
(And what limitations I might be overseeing for this to work)

**Comment by lucsmachado** [comment]
I'm also very interested in this use case!
Have you been able to come up with a working example of what a `<ResponsiveDialog />` implementation might look like?

**Comment by sourcebert** [comment]
There is already an attempt to do this: https://github.com/redpangilinan/credenza
